agent/agent.go

```diff
@@ -318,3 +327,28 @@ func initializeStateManager(cfg *config.Config, taskEngine engine.TaskEngine, cl
 	}
 	return stateManager, nil
 }
+
+// initializeENIWatcher wraps up the setup for creating the ENI Watcher
+func initializeENIWatcher(ctx context.Context) (*eniWatcher.UdevWatcher, error) {
```

Comment:
Can this be moved somewhere other than `agent.go`?  It would be good to get it into a different package and write tests around it.